data/data/powervs/loadbalancer/alb.tf

```diff
@@ -0,0 +1,151 @@
+################################################################
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Licensed Materials - Property of IBM
+#
+# ©Copyright IBM Corp. 2021
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+################################################################
+
+locals {
+  api_servers       = concat([var.bootstrap_ip], var.master_ips)
+  api_servers_count = 4 # bootstrap + master 
+  app_servers       = var.master_ips
+  app_servers_count = 3
+}
+
+resource "ibm_is_lb" "load_balancer" {
+  name            = "${var.cluster_id}-loadbalancer"
+  subnets         = [var.vpc_subnet_id]
+  security_groups = [ibm_is_security_group.ocp_security_group.id]
+}
+
+# machine config listener and backend pool
+resource "ibm_is_lb_listener" "machine_config_listener" {
+  lb           = ibm_is_lb.load_balancer.id
+  port         = 22623
+  protocol     = "tcp"
+  default_pool = ibm_is_lb_pool.machine_config_pool.id
+}
+resource "ibm_is_lb_pool" "machine_config_pool" {
+  depends_on = [ibm_is_lb.load_balancer]
```

Comment:
do we need this? because `lb = ibm_is_lb.load_balancer.id` will take care the dependecy right?

Reply:
same comment applies for all the depends_on statements

Reply:
Right, I think some of the explicit dependencies can be removed.

Reply:
The members are added one by one to the pool and if the explicit dependency is removed the code will get stuck.
So when one member is added the LB status is updating.  If we don't wait for the state to be ready before adding another member there are issues. Using the load balancer `id` doesn't guarantee this behaviour
 Ref: https://registry.terraform.io/providers/IBM-Cloud/ibm/latest/docs/resources/is_lb_listener

In short the resources needs to be added serially.

Reply:
Got it, that is the reason even in the UI it allows only one operation at a time, seems like the resource itself can handle one ops at a time.!